x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/package-info.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/**
+ * Functions that aggregate values, with or without grouping within buckets.
+ * Used in `STATS` and similar commands.
+ *
+ * <h2>Guide to adding new aggregate function</h2>
+ * <ol>
+ *     <li>
+ *         Aggregation functions are more complex than scalar functions, so it's a good idea to discuss
+ *         the new function with the ESQL team before starting to implement it.
+ *         <p>
+ *             You may also discuss its implementation, as aggregations may require special performance considerations.
+ *         </p>
+ *     </li>
+ *     <li>
+ *         Run the csv tests (see {@code x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/CsvTests.java})
```

Comment:
Maybe tell folks to go make a scalar function first. That way we can assume knowledge of that list.